pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -0,0 +1,133 @@
+package autosizednodes
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/util"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+)
+
+type Reconciler struct {
+	client client.Client
+
+	log *logrus.Entry
+}
+
+const (
+	configName                 = "dynamic-node"
+	AUTOSIZED_NAMESPACE string = "aro.autosizednodes"
+	AUTOSIZED_ENABLED   string = AUTOSIZED_NAMESPACE + ".enabled"
+)
+
+func NewReconciler(log *logrus.Entry, mgr ctrl.Manager) *Reconciler {
+	return &Reconciler{
+		client: mgr.GetClient(),
+
+		log: log,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	var aro arov1alpha1.Cluster
+	var err error
+
+	err = r.client.Get(ctx, request.NamespacedName, &aro)
+	if err != nil {
+		err = fmt.Errorf("unable to fetch aro cluster: %w", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.log.Infof("Config changed, autoSize: %t\n", aro.Spec.OperatorFlags.GetSimpleBoolean(AUTOSIZED_ENABLED))
+
+	// key is used to locate the object in the etcd
+	key := types.NamespacedName{
+		Name: configName,
+	}
+
+	if !aro.Spec.OperatorFlags.GetSimpleBoolean(AUTOSIZED_ENABLED) {
+		// defaults to deleting the config
+		config := mcv1.KubeletConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: configName,
+			},
+		}
+		err = r.client.Delete(ctx, &config, &client.DeleteOptions{})
+		if client.IgnoreNotFound(err) != nil {
+			err = fmt.Errorf("could not delete KubeletConfig: %w", err)
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
I think one of the `client.IgnoreNotFound` os not necessary:

```suggestion
		if err != nil {
			err = fmt.Errorf("could not delete KubeletConfig: %w", err)
		}
		return ctrl.Result{}, client.IgnoreNotFound(err)
```

`client.IgnoreNotFound(err)` will unwrap the error and will return `nil` if underlying error in an "not found" error.

Reply:
Actually instead of previous suggestion - I think this is better to be consistent with what we do after `r.client.Get(ctx, request.NamespacedName, &aro)`.

```suggestion
		if err != nil {
			err = fmt.Errorf("could not delete KubeletConfig: %w", err)
			return ctrl.Result{}, client.IgnoreNotFound(err)
		}
```

Reply:
This will not cover the case: Not Needed but present. With this change it will always recreate the config. So this needs to stay.